test/MemeTest.sol

```diff
@@ -129,12 +111,98 @@ contract MemeTest is TestBaseContract {
         );
 
         assertEq(memeToken.balanceOf(user1), 1000 ether);
+    }
+
+    function testFuzz_buyMemeWithMockTwap(uint256 amount) public {
+        amount = bound(amount, 1 ether, 1000 ether);
+
+        MockTwap twap = new MockTwap();
+
+        twap.setSqrtPriceX96(1625 * 10 ** 18);
+        proxy.setPriceOracle(address(twap));
 
-        // todo add more checks
-        // proxy.buyMeme{ value: 20 ether }(MemeBuySizeDollars.Ten);
-        // proxy.buyMeme{ value: 30 ether }(MemeBuySizeDollars.Twenty);
-        // proxy.buyMeme{ value: 70 ether }(MemeBuySizeDollars.Fifty);
-        // proxy.buyMeme{ value: 200 ether }(MemeBuySizeDollars.Hundred);
+        vm.deal(user1, 1e6 ether);
+
+        uint256 user1_startingBalance;
+
+        uint256 server_startingBalance;
+        uint256 treasury_startingBalance;
+
+        uint256 user1_endingBalance;
+
+        uint256 cost; // cost of meme in coin
+        uint256 buyAmount; // amount of MEME bought
+        uint256 server_endingBalance;
+        uint256 treasury_endingBalance;
+
+        MemeBuySizeDollars size;
+
+        uint256 user1_startingMemeBalance;
+        for (uint256 i; i < 5; ++i) {
+            if (i == 0) {
+                size = MemeBuySizeDollars.Five;
+            } else if (i == 1) {
+                size = MemeBuySizeDollars.Ten;
+            } else if (i == 2) {
+                size = MemeBuySizeDollars.Twenty;
+            } else if (i == 3) {
+                size = MemeBuySizeDollars.Fifty;
+            } else if (i == 4) {
+                size = MemeBuySizeDollars.Hundred;
+            }
+            user1_startingBalance = user1.balance;
+
+            server_startingBalance = server.addr.balance;
+            treasury_startingBalance = address(proxy).balance;
+
+            user1_startingMemeBalance = proxy.tokenBalanceOf(LibTokenIds.TOKEN_MEME, user1);
+            (cost, buyAmount) = proxy.getMemeCost(size);
+            if (cost <= amount) {
+                vm.prank(user1);
+                proxy.buyMeme{ value: amount }(size);
+                assertEq(
+                    proxy.tokenBalanceOf(LibTokenIds.TOKEN_MEME, user1),
+                    user1_startingMemeBalance + buyAmount,
+                    "user1 meme balance should INCREASE"
+                );
+                assertEq(
+                    memeToken.balanceOf(user1),
+                    user1_startingMemeBalance + buyAmount,
+                    "user1 meme balance should INCREASE"
+                );
+            } else {
+                delete cost;
+            }
+
+            user1_endingBalance = user1.balance;
+
+            server_endingBalance = server.addr.balance;
+            treasury_endingBalance = address(proxy).balance;
+
+            if (server_startingBalance > LibConstants.SERVER_COIN_TRESHOLD) {
```

Comment:
Let's put this into its own simple test, keep the fuzzing test just about purchasing MEME.